csbmysql/resource_binding_user_test.go

```diff
@@ -241,15 +241,8 @@ func checkSSLCipher(requireSSL bool) resource.TestCheckFunc {
 		Expect(err).NotTo(HaveOccurred())
 		Expect("Ssl_cipher").To(Equal(res.VariableName))
 
-		var expectedCipher string
-		switch getMySQLVersion() {
-		case latestMySQLVersion:
-			expectedCipher = "TLS_AES_128_GCM_SHA256"
-			Expect(expectedCipher).To(Equal(res.Value))
-		default:
-			expectedCipher = "AES128-GCM-SHA256"
-			Expect(expectedCipher).To(Equal(res.Value))
-		}
+		expectedCiphers := []string{"TLS_AES_128_GCM_SHA256", "AES128-GCM-SHA256", "ECDHE-RSA-AES128-GCM-SHA256"}
+		Expect(expectedCiphers).To(ContainElement(res.Value))
```

Comment:
I think there's a nicer way to write this:
```go
Expect(res.Value).To(SatisfyAny(
  Equal("TLS_AES_128_GCM_SHA256"),
  Equal("AES128-GCM-SHA256"),
  Equal("ECDHE-RSA-AES128-GCM-SHA256"),
))
```
What I prefer about this is that the expectation is the right way around and it reads like a sentence.

Reply:
Hi @blgm, thanks for the suggestion.
I was aware of both possibilities and in this case I preferred using the ContainElement matcher.

However, while I was trying to justify my decision I understood your point.
The current implementation would return a very misleading error message. Such as:

> Expected {"TLS_AES_128_GCM_SHA256", "AES128-GCM-SHA256", "ECDHE-RSA-AES128-GCM-SHA256"} to contain element matching "SOME_TLS_ALGORITHM"

Which is an absolute lie. What we were really expecting is the following or something similar:

> Expected SOME_TLS_ALGORITHM To satisfy at least one of these matchers:
> SOME_TLS_ALGORITHM to equal TLS_AES_128_GCM_SHA256
> SOME_TLS_ALGORITHM to equal AES128-GCM-SHA256
> SOME_TLS_ALGORITHM to equal ECDHE-RSA-AES128-GCM-SHA256

And that's exactly what the output of your suggested changes would look like.
I'll commit a fix for it.
Thank you for your careful review and for your suggestion.